### PR TITLE
Update to the acquire/update/commit api

### PR DESCRIPTION
**Pull request by vmpstr**
Please take a look.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,12 +34,10 @@ subtree are not rendered immediately. Instead, when the lock is committed, the
...
-The visual content of an element's subtree which is locked for display does not
-change. Specifically, if the element already existed in the DOM, then the
-content that was present at the time the lock was acquired remains visibile
-until the lock is committed and the associated promise is resolved. Note that in
-this case, the element itself still responds and updates its own visual state
-(e.g. border); it is only the element's subtree that is locked for display.
+The current consideration is only for an element to be locked before it is
+inserted into the DOM. Display locking can later be extended to lock elements
+already in the DOM, either freezing their visual state, or making them
+invisible.
```
Comment (chrishtr) [inline README.md]: We will need the "make them invisible" part soon for unification with Searchable Invisible DOM. Just leave
this paragraph as-is?

File: README.md [hunk 2]
```diff
@@ -48,46 +46,52 @@ display any content for this element or its subtree until the work is completed.
...
+  let element = document.createElement("div");
+
+  // Construct the element
+  element.appendChild(...);
+  element.id = "...";
+
+  // Acquire a lock.
+  await element.getDisplayLock().acquire();
+
+  // Append the element to the DOm.
```
Comment (chrishtr) [inline README.md]: Typo: DOM

File: README.md (hunk 1 above)
Comment (vmpstr) [reply README.md]: Done. I removed a small detail, since it doesn't seem relevant.

File: README.md (hunk 2 above)
Comment (vmpstr) [reply README.md]: Done.

**Comment by vmpstr** [comment]
PTAL

File: explainer.md [hunk 3]
```diff
@@ -230,145 +240,99 @@ The display locking proposal is intended to improve the script, layout, paint,
...
-##### Element.acquireDisplayLock()
+##### Element.getDisplayLock()
 
 With display locking, each of the `Element` objects has a new function,
-`acquireDisplayLock()` which takes a callback to run when the lock is acquired
-and returns a promise which resolves when all of the work to layout, and paint
-the subtree has completed. The callback given to `acquireDisplayLock()` takes
-a DisplayLockContext object, which provides functionality for more sophisticated
-designs.
+`getDisplayLock()` which returns a DisplayLockContext reprsenting the display
```
Comment (chrishtr) [inline explainer.md]: Nit: representing

File: explainer.md [hunk 4]
```diff
@@ -230,145 +240,99 @@ The display locking proposal is intended to improve the script, layout, paint,
...
+lock. The rest of the display locking function happens by interacting with this
```
Comment (chrishtr) [inline explainer.md]: s/function/functionality/

File: explainer.md [hunk 5]
```diff
@@ -230,145 +240,99 @@ The display locking proposal is intended to improve the script, layout, paint,
...
+
+Note that the acquire call takes options, consisting of the following key-value
+pairs:
+* (optional) timeout: timeoutInMilliseconds
+  * This is an optional timeout value for the duration of the lock. When the
+    lock is acquired, and commit is not called within timeoutInMilliseconds
+    milliseconds, the lock will be automatically committed. This is a safety
+    valve for protecting pages from inadvertently leaving an element locked
+    indefinitely due to uncaught exceptions or other bugs in the code. For a
+    more sofisticated use cases, the value of Infinity will effectively disable
```
Comment (chrishtr) [inline explainer.md]: sophisticated

File: explainer.md (hunk 3 above)
Comment (vmpstr) [reply explainer.md]: Done.

File: explainer.md (hunk 4 above)
Comment (vmpstr) [reply explainer.md]: Done.

File: explainer.md (hunk 5 above)
Comment (vmpstr) [reply explainer.md]: Done.

**Comment by vmpstr** [comment]
PTAL

File: explainer.md [hunk 6]
```diff
@@ -26,17 +26,14 @@ previous layout and visual update to be done before running subsequent script.
...
-as running script or DOM updates outside of the locked subtree. The developer
-will then be able to unlock the element, causing the visual updates of the
-modified subtree to appear. Note that unlock would still be asynchronous,
-meaning that the request to unlock can take some time to process to allow for
-co-operative updates to complete. In essence, display locking will make it
-possible to perform complicated DOM updates without causing the rest of the page
-to jank. However, it will also come at a small cost: DOM updates in a locked
-subtree will not be visually updated until the lock is committed, and input
-events targeted for the locked subtree will not be processed.
+will be able construct the locked subtree’s DOM however it desires, and insert
```
Comment (chrishtr) [inline explainer.md]: "however they desire"

File: explainer.md [hunk 7]
```diff
@@ -230,145 +240,99 @@ The display locking proposal is intended to improve the script, layout, paint,
...
 With display locking, each of the `Element` objects has a new function,
-`acquireDisplayLock()` which takes a callback to run when the lock is acquired
-and returns a promise which resolves when all of the work to layout, and paint
-the subtree has completed. The callback given to `acquireDisplayLock()` takes
-a DisplayLockContext object, which provides functionality for more sophisticated
-designs.
+`getDisplayLock()` which returns a DisplayLockContext representing the display
+lock. The rest of the display locking functionality happens by interacting with this
+object. The DisplayLockContext is bound to the element from which it was
+retrieved, meaning that operations on the object will affect the source element.
```
Comment (chrishtr) [inline explainer.md]: s/the source/that/

File: explainer.md (hunk 6 above)
Comment (vmpstr) [reply explainer.md]: Done.

File: explainer.md (hunk 7 above)
Comment (vmpstr) [reply explainer.md]: Done.

**Pull request merged**